trieste/models/gpflow/sampler.py

```diff
@@ -49,26 +49,29 @@
 
 
 def qmc_normal_samples(
-    num_samples: _IntTensorType, n_sample_dim: _IntTensorType, skip: _IntTensorType = 0
+    num_samples: _IntTensorType,
+    n_sample_dim: _IntTensorType,
+    skip: _IntTensorType = 0,
+    dtype: tf.DType = tf.float64,
 ) -> tf.Tensor:
     """
     Generates `num_samples` sobol samples, skipping the first `skip`, where each
     sample has dimension `n_sample_dim`.
     """
 
     if num_samples == 0 or n_sample_dim == 0:
-        return tf.zeros(shape=(num_samples, n_sample_dim), dtype=tf.float64)
+        return tf.zeros(shape=(num_samples, n_sample_dim), dtype=dtype)
```

Comment:
I think this path isn't tested... worth adding a case for `n_sample_dim==0` in the unit tests?